### PR TITLE
doubled wall spacing

### PR DESCRIPTION
**Pull request by piman51277**
Completes RC SW Tutorials.

### CODE REVIEW

**Comment by piman51277** [comment]
image

**Comment by piman51277** [comment]
Evidence of topic publish:
bananas
blueberries

**Comment by piman51277** [comment]
Feedback:
I liked the tutorial, but it should have included the note about adding to cmake when we had to make a new position.

File: soccer/src/soccer/strategy/agent/position/runner.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+std::optional<RobotIntent> Runner::derived_get_task(RobotIntent intent) {
+    latest_state_ = update_state();
+    return state_to_task(intent);
+}
+
+std::string Runner::get_current_state() { return "Runner"; }
+
+bool inPosition(rj_geometry::Point current, rj_geometry::Point target, double tolerance) {
+    return current.dist_to(target) < tolerance;
```
Comment (jacksherling) [inline soccer/src/soccer/strategy/agent/position/runner.cpp:19]: take a look at the `check_is_done` function. When the plan succeeds it returns true. This way thresholding for plans is abstracted away from strategy

File: soccer/src/soccer/strategy/agent/position/runner.cpp [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        intent.motion_command = mark_cmd;
+    }
+    return intent;
+
+    // should be impossible to reach, but this is equivalent to
+    // sending an empty MotionCommand
+    return std::nullopt;
+}
+
+bool Runner::shot_on_goal_detected(WorldState* world_state) {
```
Comment (jacksherling) [inline soccer/src/soccer/strategy/agent/position/runner.cpp:71]: Looks like you left some extra stuff from another position here

File: soccer/src/soccer/strategy/agent/position/runner.hpp [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    // see Position superclass
+    std::optional<RobotIntent> derived_get_task(RobotIntent intent) override;
+
+    // possible states of the Runner
+    enum State {
+        IDLING,          // doing nothing
+        BLOCKING,        // blocking the ball from reaching the goal
+        CLEARING,        // clearing the ball out of the goal box
+        PREPARING_SHOT,  // pivot around ball in preparation for shot
```
Comment (jacksherling) [inline soccer/src/soccer/strategy/agent/position/runner.hpp:50]: Similar to above, let's remove stuff not used for runner

File: soccer/src/soccer/soccermom/soccermom.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                auto message = std_msgs::msg::String();
+                message.data = "bananas";
+                team_fruit_pub_->publish(message);
+            }
+
+        });
+
+    team_fruit_pub_ = create_publisher<std_msgs::msg::String>("/team_fruit", rclcpp::QoS(10));
+
+    if (blue_team) {
```
Comment (jacksherling) [inline soccer/src/soccer/soccermom/soccermom.cpp:33]: I think this work is already done in the subscription

File: soccer/src/soccer/soccermom/soccermom.cpp [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    : Node{"soccermom", rclcpp::NodeOptions{}
+                            .automatically_declare_parameters_from_overrides(true)
+                            .allow_undeclared_parameters(true)}
+
+{
+    team_color_sub_ = create_subscription<rj_msgs::msg::TeamColor>(
+        referee::topics::kTeamColorTopic, rclcpp::QoS(1).transient_local(),
+        [this](rj_msgs::msg::TeamColor::SharedPtr color) {  // NOLINT
+            team_fruit_pub_ =
+                create_publisher<std_msgs::msg::String>("/team_fruit", rclcpp::QoS(10));
```
Comment (jacksherling) [inline soccer/src/soccer/soccermom/soccermom.cpp:15]: why create a new publisher on every subscription callback?

File: soccer/src/soccer/soccermom/soccermom.hpp [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    Soccermom(bool blue_team = false);
+
+protected:
+    // Time between consecutive calls to tick().
+    std::chrono::milliseconds tick_period_ = std::chrono::milliseconds(100);
+    // Ros timer to trigger tick every tick_period
+    rclcpp::TimerBase::SharedPtr tick_timer_;
+
+    // The position of each robot
+    std::array<strategy::Positions, kNumShells> positions_;
```
Comment (jacksherling) [inline soccer/src/soccer/soccermom/soccermom.hpp:46]: Let's remove the unused variables here

**Review by jacksherling — CHANGES_REQUESTED** [review]
Great job overall!

Please review the comments and close this PR when you're finished (no need to actually make the changes for the tutorial)! Feel free to let me know if you have any questions.

**Review by jacksherling — APPROVED** [review]

**Pull request closed without merging**